Content.Shared/RCD/Systems/RCDAmmoSystem.cs

```diff
@@ -42,6 +42,16 @@ private void OnAfterInteract(EntityUid uid, RCDAmmoComponent comp, AfterInteract
             return;
 
         var user = args.User;
+
+        // ## Frontier - Shipyard RCD ammo only fits in shipyard RCD.
+        // At this point RCDComponent is guaranteed
+        EnsureComp<RCDComponent>(target, out var rcdComponent);
```

Comment:
optionally you could change the HasComp above to a try comp and out the rcdComponent, but for the sake of code seperation this is fine, and its not called often anyway to worry about query calls.